catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/osgi/OsgiServices.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor.osgi;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Helper class for interacting with OSGi services. */
+public class OsgiServices {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(OsgiServices.class);
+
+  private static final String INVALID_LDAP_FILTER_SYNTAX = "Invalid LDAP filter syntax";
+
+  public BundleContext bundleContextFor(Class aClass) {
+    Bundle bundle = FrameworkUtil.getBundle(aClass);
```

Comment:
❓ Why not just inject the bundle context instead? It doesn't look like you're taking advantage of the dynamic class lookup since all this code is in the same bundle.